src/basic/strv.c

```diff
@@ -133,43 +133,28 @@ char **strv_new_ap(const char *x, va_list ap) {
          * STRV_IFNOTNULL() macro to include possibly NULL strings in
          * the string list. */
 
-        if (x) {
-                n = x == STRV_IGNORE ? 0 : 1;
-
-                va_copy(aq, ap);
-                while ((s = va_arg(aq, const char*))) {
-                        if (s == STRV_IGNORE)
-                                continue;
-
-                        n++;
-                }
+        va_copy(aq, ap);
+        for (s = x; s; s = va_arg(aq, const char*)) {
```

Comment:
Here too, it'd be nicer to define `s` here, and separately in the loop below.

Reply:
OK, I was actually going back and forth on that; if there'd been only one loop it would have been an easier decision. Anyway, I'll wait a few days for other comments, then update with those changes.